### PR TITLE
Chatting Working for the TopRightIcon

### PR DESCRIPTION
**Pull request by jasonbot123**

### CODE REVIEW

File: src/main/java/view/ChatPageUI/ChatPage.java [hunk 1]
```diff
@@ -54,14 +54,14 @@ public ChatPage(JFrame parentFrame) {
...
-                    ChatWindow chatWindow = new ChatWindow(selectedUser);
+                    ChatWindow chatWindow = new ChatWindow(selectedUser, username);
                     chatWindow.setVisible(true);
                 }
             }
         });
 
         // Load chat contacts for the current user
-        chatContactController.getChatContacts("jason");
+        chatContactController.getChatContacts(username);
```
Comment (CelineChien81777) [inline src/main/java/view/ChatPageUI/ChatPage.java:64]: Uses the username constructor, good change

File: src/main/java/view/HomePageUI/TopRightIconsPanel.java [hunk 2]
```diff
@@ -50,24 +51,27 @@ public TopRightIconsPanel(JFrame parentFrame) {
                         );
                     }
                 });
+            }
+            if ("📩".equals(icon)) {
+                iconButton.addActionListener(e -> {
+                    String sectionFilter = resolveSectionFilter(parentFrame);
+                    if (sectionFilter != null) {
+                        ChatPage chatpage = new ChatPage(parentFrame, currentuser);
```
Comment (CelineChien81777) [inline src/main/java/view/HomePageUI/TopRightIconsPanel.java:59]: the action listener follows a good coding style, it is clean and tidy.

File: src/main/java/view/ChatPageUI/ChatPage.java [hunk 3]
```diff
@@ -21,7 +21,7 @@
 public class ChatPage extends JFrame {
     private final DefaultListModel<String> contactsListModel;
 
-    public ChatPage(JFrame parentFrame) {
```
Comment (CelineChien81777) [inline src/main/java/view/ChatPageUI/ChatPage.java:24]: No javadocs

File: src/main/java/view/ChatPageUI/MockChatViewApp.java [hunk 4]
```diff
@@ -44,9 +44,9 @@ public static void main(String[] args) {
         ChatContactInteractor interactor = new ChatContactInteractor(contactPresenter, chatDataAccess);
         ChatContactController chatContactController = new ChatContactController(interactor);
 
-        JFrame frame = new JFrame();
-        ChatPage chatPage = new ChatPage(frame);
-        chatPage.setVisible(true);
+//        JFrame frame = new JFrame();
```
Comment (CelineChien81777) [inline src/main/java/view/ChatPageUI/MockChatViewApp.java:47]: Should delete the unused lines

File: src/main/java/view/ChatPageUI/ChatWindow.java [hunk 5]
```diff
@@ -19,13 +19,14 @@
 
 public class ChatWindow extends JFrame {
     private final ChatController chatController;
-    private final String currentUser = "jason";
+    private final String currentUser;
```
Comment (CelineChien81777) [inline src/main/java/view/ChatPageUI/ChatWindow.java:22]: Uses the correct logic where the username should be the user that is currently logged in, not a placeholder.

File: src/main/java/view/HomePageUI/TopRightIconsPanel.java (hunk 2 above)
Comment (JanosRenAI) [reply src/main/java/view/HomePageUI/TopRightIconsPanel.java:59]: There are two constructors, one using parentFrame and another using both parentFrame and currentuser. If the first constructor is not being used, consider removing it to simplify the code.

**Pull request merged**